autoload/OmniSharp/locations.vim

```diff
@@ -55,17 +55,61 @@ function! OmniSharp#locations#Preview(location) abort
   endif
 endfunction
 
-function! OmniSharp#locations#SetQuickfix(list, title)
+function! OmniSharp#locations#SetQuickfix(list, title) abort
+  call s:SetQuickfixFromDict(a:list, {'title': a:title})
+endfunction
+
+function! OmniSharp#locations#SetQuickfixWithVerticalAlign(list, title) abort
+  call s:SetQuickfixFromDict(a:list, {'title': a:title, 'quickfixtextfunc': 's:QuickfixVerticalAlignFunction'})
+endfunction
+
+function! s:SetQuickfixFromDict(list, dict) abort
   if !has('patch-8.0.0657')
-  \ || setqflist([], ' ', {'nr': '$', 'items': a:list, 'title': a:title}) == -1
-    call setqflist(a:list)
+  \ || setqflist([], ' ', {'nr': '$', 'items': a:list, 'title': get(a:dict, 'title', 'OmniSharp')}) == -1
+    call setqflist(a:list, ' ', a:dict)
   endif
   silent doautocmd <nomodeline> QuickFixCmdPost OmniSharp
   if g:OmniSharp_open_quickfix
     botright cwindow
   endif
 endfunction
 
+function! s:QuickfixVerticalAlignFunction(info) abort
+	if a:info.quickfix
+		let qfl = getqflist({'id': a:info.id, 'items': 0}).items
+	else
+		let qfl = getloclist(a:info.winid, {'id': a:info.id, 'items': 0}).items
+	endif
+	let l = []
+	let efm_type = {'e': 'error', 'w': 'warning', 'i': 'info', 'n': 'note'}
+	let lnum_width =   len(max(map(range(a:info.start_idx - 1, a:info.end_idx - 1), { _,v -> qfl[v].lnum })))
+	let col_width =    len(max(map(range(a:info.start_idx - 1, a:info.end_idx - 1), {_, v -> qfl[v].col})))
+	let fname_width =  max(map(range(a:info.start_idx - 1, a:info.end_idx - 1), {_, v -> strchars(fnamemodify(bufname(qfl[v].bufnr), ':t'), 1)}))
+	let type_width =   max(map(range(a:info.start_idx - 1, a:info.end_idx - 1), {_, v -> strlen(get(efm_type, qfl[v].type, ''))}))
+	let errnum_width = len(max(map(range(a:info.start_idx - 1, a:info.end_idx - 1),{_, v -> qfl[v].nr})))
+	for idx in range(a:info.start_idx - 1, a:info.end_idx - 1)
+		let e = qfl[idx]
+		if !e.valid
+			call add(l, '|| ' .. e.text)
+		else
```

Comment:
`..` concatenation is not available in older vims, or neovim. I realise that this function won't get called in those situations anyway, but this will light up as an error in linters. Also, I'd prefer to remain consistent with the rest of the codebase.

Reply:
No problem